base/static/base/js/jquery.track-everything.js

```diff
@@ -276,17 +317,105 @@
+                        }
+                    } else {
+                        // Remove option
+                        currentOption = currentOption.split('; ').filter(opt => opt !== optionText).join('; ');
+                    }
+                    searchButton.setAttribute('data-ga-event-option', currentOption);
+                });
+            }
+
+        } else
+            if (target.matches('input[type="radio"]') && target.closest(SELECTORS.SEARCH_WIDGET)) {
+                const searchWidget = target.closest(SELECTORS.SEARCH_WIDGET);
+                const searchButtons = searchWidget.querySelectorAll('button.btn-search[type="submit"][data-ga-subcategory="' + target.getAttribute('data-ga-subcategory') + '"]');
+                if (searchButtons.length) {
+                    searchButtons.forEach((searchButton) => {
+                        let optionText = target.getAttribute('data-ga-label');
+                        let currentOption = searchButton.getAttribute('data-ga-event-option') || '';
+                        let optionsArr = currentOption ? currentOption.split('; ') : [];
+                        // Remove any previous rad:... entry
+                        optionsArr = optionsArr.filter(opt => !opt.startsWith('radio:'));
+                        if (target.checked) {
+                            optionsArr.push('radio:' + optionText);
+                        }
+                        currentOption = optionsArr.join('; ');
+                        searchButton.setAttribute('data-ga-event-option', currentOption);
+                    });
+                }
+
+            } else
+                // Selectpicker in search-widget
+                if (target.matches('[role="tabpanel"][data-ga-subcategory="tab-catalog"] select.selectpicker.btn-searchtype') && target.closest(SELECTORS.SEARCH_WIDGET)) {
+                    const searchWidget = target.closest(SELECTORS.SEARCH_WIDGET);
+                    const searchButton = searchWidget.querySelector('button[type="submit"].btn-search');
+                    if (searchButton) {
+                        // let selectedValue = target.options[target.selectedIndex].text.trim();
+                        let selectedValue = target.value.trim();
+                        let currentOption = searchButton.getAttribute('data-ga-event-option') || '';
+                        let optionsArr = currentOption ? currentOption.split('; ') : [];
+                        // Remove any previous type:... entry
+                        optionsArr = optionsArr.filter(opt => !opt.startsWith('type:'));
+                        if (selectedValue !== 'AllFields') {
+                            optionsArr.push('type:' + selectedValue);
+                        }
+                        currentOption = optionsArr.join('; ');
+                        searchButton.setAttribute('data-ga-event-option', currentOption);
+                    }
+                }

```

Comment:
The logic for these could be refactored into handle functions, that could be called like so:
```
  function handleOptionChange(e) {
      const target = e.target;
      const searchWidget = target.closest(SELECTORS.SEARCH_WIDGET);
      if (!searchWidget) return;

      if (target.matches('input[type="checkbox"]')) {
          handleCheckboxChange(target, searchWidget);
      } else if (target.matches('input[type="radio"]')) {
          handleRadioChange(target, searchWidget);
      } else if (target.matches('select.selectpicker.btn-searchtype')) {
          handleSelectChange(target, searchWidget);
      }
  }
```

That might make this a little more maintainable. Not a big deal though.